Sources/BigInt/BigInt.swift

```diff
@@ -0,0 +1,816 @@
+//===--- BigInt.swift -----------------------------------------*- swift -*-===//
+//
+// This source file is part of the Swift Numerics open source project
+//
+// Copyright (c) 2019 - 2020 Apple Inc. and the Swift Numerics project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+public struct BigInt: SignedInteger {
+
+  public typealias Words = [UInt]
+
+  public private(set) var words: Words
+
+  @usableFromInline
+  internal init(words: Words) {
+    self.words = words
```

Comment:
I think there is a normalization step missing here -- e.g., `BigInt(words: [0, 0, 0, 0])` should result in an integer with a single digit.

Reply:
The idea with this particular initializer is that it's for internal (really private, but internal is required for `@usableFromInline`) use only, and the words in question should be pre-normalized. So the options here are add a comment to that effect, or risk the slowdown from checking if we need to do normalization here. I'm open to either one, in most cases the check is just going to be false so it doesn't add a huge amount of overhead.

Reply:
Oh, my apologies, I somehow missed that this is `@usableFromInline` declaration.

The stdlib convention is to put a leading underscore on all internal/private names (whether or not they are also @usableFromInline), to prevent conflicts in case we want to introduce public API with the same name later. (It would be a shame to be forced to choose a suboptimal name because the good name is already taken by an ABI-stable implementation detail.) For initializers, this means putting an underscore on the first argument label, as in `init(_words:)`.

Additionally, we prefer to put `unchecked` (sometimes even `unsafe`) in the name of internal initializers that skip expected normalization/verification steps, to highlight their unusual behavior.

So per the stdlib conventions, the recommended name for this initializer would be something like this:

```swift
 @usableFromInline
  internal init(_uncheckedWords words: Words) { ... }
```

Reply:
Should there also be a public initializer? For example:

```swift
public init<T>(
  words: T
) where T: RandomAccessCollection, T.Element == UInt, T.Index == Int
```

This would replace the existing `init(bitPattern:)` below (lines 23 to 25), which is unlike the standard library initializers in [IntegerTypes.swift.gyb][].

[IntegerTypes.swift.gyb]: <https://github.com/apple/swift/blob/8510b9234b35938b2869041a9286bd7a0e3bb1c5/stdlib/public/core/IntegerTypes.swift.gyb#L1102-L1115>

Reply:
Definitely! But an initializer like that would also be useful as a `BinaryInteger` requirement, so to prevent source compatibility headaches, we should consider adding it to the stdlib before we add it to SwiftNumerics.

swift-evolution can litigate the exact spelling & constraints, and BigInt can adopt whatever gets accepted.